src/store/config/getters.ts

```diff
@@ -90,25 +90,27 @@ export const getters: GetterTree<ConfigState, RootState> = {
 
   getMergedTableHeaders: (state, getters) => (headers: AppTableHeader[], key: string) => {
     const configured: AppTablePartialHeader[] = getters.getConfiguredTableHeaders(key)
+
     if (!configured) {
       return headers
     }
-    const merged: AppTableHeader[] = []
-    headers.forEach(header => {
-      const keyBy = (header.key)
-        ? 'key'
-        : 'value'
-
-      const o = {
-        visible: true,
-        configurable: false,
-        ...header,
-        ...configured.find(p => p[keyBy] === header[keyBy])
-      }
 
-      merged.push(o)
-    })
-    return merged
+    // if the number of configured headers is the same as the available headers,
+    // then use configured headers to keep order
+    // else go with available headers
+    return headers.length === configured.length
+      ? configured
+        .map(header => ({
+          visible: true,
+          ...headers.find(p => p.value === header.value),
+          ...header,
+        }))
+      : headers
+        .map(header => ({
+          visible: true,
+          ...header,
+          ...configured.find(p => p.value === header.value)
+        }))
```

Comment:
I added some logs locally and noticed my configured header count was the same as the available header count. I played around with the merging code here and this seems to work reliable for me:
```suggestion
    const configuredHeaderValues = configured.map(header => header.value);
    const allHeaders = [...configured];
    for(const header of headers)
      if(!configuredHeaderValues.includes(header.value))
        allHeaders.push(header);

    return allHeaders.map(header => ({
      visible: true,
      ...headers.find(p => p.value === header.value), // apply original values
      ...header, // apply overrides
    }))
```

Reply:
If the header count is the same, how would this change make a difference? Something odd is going on here... can you check what you have on the database?

Reply:
My guess is that you had some weird error on the database, possibly from the development phase of the column picker code for Spoolman, and this just brought it up!

I've tried in both my printers and even with a brand new database and seems to work fine!

Reply:
Sounds good to me then ;)